apps/docs/src/pages/en/features/completion.md

```diff
@@ -0,0 +1,172 @@
+---
+title: CompletionFactory
+layout: ../../../layouts/MainLayout.astro
+---
+
+So for now, we've got a CLI with all the needed tools. The problem is - the end
+user does not know what commands are available. We can fix that with the
+`CompletionFactory`! This service has the ability to generate a completion
+script for the end user to use, both for ZSH & Bash and Fig! The completion will
+be generated dynamically based on the commands and options available.
+
+# Setup Your Completion Using The CompletionFactory
+
+Your `main.ts` file:
+
+```typescript
+import { CommandFactory, CompletionFactory } from 'nest-commander';
+import { AppModule } from './app.module';
+
+(async () => {
+  const cliCommand = 'YOUR-CLI-COMMAND';
+  const cliExecutable = 'YOUR-CLI-EXECUTABLE-PATH';
+
+  const app = await CommandFactory.createWithoutRunning(AppModule, {
+    completion: {
+      cmd: cliCommand,
+      fig: true,
+      nativeShell: {
+        executablePath: cliExecutable
+      }
+    }
+  });
```

Comment:
To be clear as well, this can just be `await CommandFactory.run(AppModule, optionsWithCompletions)`, yeah?

Reply:
Indeed, your code built in a way that the creating of the application logic happens under the hood just the same.

Reply:
Perfect, that's exactly what I was hoping for! Great, then I think this is ready to be merged in! Thanks for the help on the feature and building up some docs around it :)